connectors/spark/src/test/java/io/unitycatalog/spark/BaseTableReadWriteTest.java

```diff
@@ -0,0 +1,195 @@
+package io.unitycatalog.spark;
+
+import static io.unitycatalog.server.utils.TestUtils.CATALOG_NAME;
+import static io.unitycatalog.server.utils.TestUtils.SCHEMA_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.unitycatalog.client.ApiException;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public abstract class BaseTableReadWriteTest extends BaseSparkIntegrationTest {
```

Comment:
The intention that you make this abstraction is ?  I guess you want to extend this `BaseTableReadWriteTest` to test the managed table ?  But for managed table creation,  it shouldn't test the external table creation cases ( I mean all those cases are non-managed table testing, then why need to abstract them).

Reply:
Basically, think about a better hierarchy.
base classes / interfaces
- BaseTableReadWriteTest (works for all tables formats and types)
- AdditionalDeltaTests (extra tests applicable Delta tables only, but both external and managed) 

real tests
- ParquetTableReadWriteTest extends BaseTableReadWriteTest
- ExternalDeltaTableReadWriteTest extend BaseTableReadWriteTest with AdditionalDeltaTests
- ManagedDeltaTableReadWriteTest extend BaseTableReadWriteTest with AdditionalDeltaTests

each test can have additional tests specific for that case.

and any additional tests that is independent of the format at all (like list tables) should be in a test.

Reply:
let me think about it a little. i understand java's limitation makes this hard.

Reply:
okay.. i actually like this - base, external(base), delta(external), parquet(external), delta_managed(base)

but for that isnt better to name all the methods to be independent of delta?